### PR TITLE
REM-73

### PR DESCRIPTION
**Pull request by imtithalhalalo**
fixing mc-currency-mapping bugs
1- in new mode after save info button should be enable
2- transaction log is empty
3- record id is empty
4- edit any record click “clear” nothing happen

### CODE REVIEW

File: src/pages/mc-currency-mapping/forms/MultiCurrencyForm.js [hunk 1]
```diff
@@ -14,55 +14,66 @@ import { SystemRepository } from 'src/repositories/SystemRepository'
...
+        formik.setFieldValue(
+          'recordId',
+          String(obj.currencyId * 1000 + obj.rateTypeId)
+        )
+        invalidate()
+      } catch (error) {}
     }
   })
+
+  const editMode = !!formik.values.recordId || !!recordId
```
Comment (RebeccaFrem02) [inline src/pages/mc-currency-mapping/forms/MultiCurrencyForm.js:62]: why are we using !!formik.values.recordId?

File: src/pages/mc-currency-mapping/index.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { useState, useContext } from 'react'
+import { useContext } from 'react'
 import toast from 'react-hot-toast'
 import Table from 'src/components/Shared/Table'
```
Comment (omarZrayka23) [inline src/pages/mc-currency-mapping/index.js:3]: remove  const invalidate = useInvalidate({
    endpointId: MultiCurrencyRepository.McExchangeMap.page
  }).  from index and put invalidate in useResourceQuery

File: src/pages/mc-currency-mapping/index.js [hunk 3]
```diff
@@ -67,27 +67,31 @@ const MultiCurrencyMapping = () => {
...
       props: {
         labels: _labels,
-        currencyId: currencyId ? currencyId : null,
-        rateTypeId: rateTypeId ? rateTypeId : null,
+        record,
         maxAccess: access,
+        recordId: record
+          ? String(record.currencyId * 1000 + record.rateTypeId)
+          : null,
         invalidate: invalidate
```
Comment (omarZrayka23) [inline src/pages/mc-currency-mapping/index.js:95]: remove invalidate

File: src/pages/mc-currency-mapping/forms/MultiCurrencyForm.js [hunk 4]
```diff
@@ -14,55 +14,66 @@ import { SystemRepository } from 'src/repositories/SystemRepository'
...
-
   const invalidate = useInvalidate({
     endpointId: MultiCurrencyRepository.McExchangeMap.page
   })
 
   const { formik } = useForm({
     initialValues: {
-      currencyId: currencyId || null,
-      rateTypeId: rateTypeId || null,
+      recordId: recordId || null,
```
Comment (omarZrayka23) [inline src/pages/mc-currency-mapping/forms/MultiCurrencyForm.js:27]: remove || null

File: src/pages/mc-currency-mapping/forms/MultiCurrencyForm.js [hunk 5]
```diff
@@ -90,7 +101,7 @@ export default function MultiCurrencyForm({ labels, maxAccess, currencyId, rateT
                 required
                 maxAccess={maxAccess}
                 onChange={(event, newValue) => {
-                  formik && formik.setFieldValue('currencyId', newValue?.recordId)
+                  formik && formik.setFieldValue('currencyId', newValue?.recordId || null)
```
Comment (omarZrayka23) [inline src/pages/mc-currency-mapping/forms/MultiCurrencyForm.js:104]: can remove null and put ''

File: src/pages/mc-currency-mapping/forms/MultiCurrencyForm.js [hunk 6]
```diff
@@ -111,7 +122,7 @@ export default function MultiCurrencyForm({ labels, maxAccess, currencyId, rateT
                 maxAccess={maxAccess}
                 readOnly={editMode}
                 onChange={(event, newValue) => {
-                  formik && formik.setFieldValue('rateTypeId', newValue?.recordId)
+                  formik && formik.setFieldValue('rateTypeId', newValue?.recordId || null)
```
Comment (omarZrayka23) [inline src/pages/mc-currency-mapping/forms/MultiCurrencyForm.js:125]: can remove null and put ''

**Comment by omarZrayka23** [comment]
Screenshot 2024-08-27 at 10 11 11 AM
required and readonly after i write in id

File: src/pages/mc-currency-mapping/forms/MultiCurrencyForm.js [hunk 7]
```diff
@@ -14,55 +14,66 @@ import { SystemRepository } from 'src/repositories/SystemRepository'
...
   const invalidate = useInvalidate({
     endpointId: MultiCurrencyRepository.McExchangeMap.page
   })
 
   const { formik } = useForm({
     initialValues: {
-      currencyId: currencyId || null,
-      rateTypeId: rateTypeId || null,
+      recordId,
+      currencyId: null,
```
Comment (omarZrayka23) [inline src/pages/mc-currency-mapping/forms/MultiCurrencyForm.js:28]: remove  null

**Review by omarZrayka23 — CHANGES_REQUESTED** [review]

File: src/pages/mc-currency-mapping/forms/MultiCurrencyForm.js [hunk 8]
```diff
@@ -14,56 +14,65 @@ import { SystemRepository } from 'src/repositories/SystemRepository'
...
-
   const invalidate = useInvalidate({
     endpointId: MultiCurrencyRepository.McExchangeMap.page
   })
 
   const { formik } = useForm({
     initialValues: {
-      currencyId: currencyId || null,
-      rateTypeId: rateTypeId || null,
+      recordId,
```
Comment (Alinadimmokdad) [inline src/pages/mc-currency-mapping/forms/MultiCurrencyForm.js:27]: recordId : recordId || null

File: src/pages/mc-currency-mapping/forms/MultiCurrencyForm.js [hunk 9]
```diff
@@ -14,56 +14,65 @@ import { SystemRepository } from 'src/repositories/SystemRepository'
...
     }
   })
+
+  const editMode = !!formik.values.recordId
+
   useEffect(() => {
     ;(async function () {
       try {
-        if (rateTypeId && currencyId) {
+        if (record.rateTypeId && record.currencyId && recordId) {
```
Comment (Alinadimmokdad) [inline src/pages/mc-currency-mapping/forms/MultiCurrencyForm.js:67]: && formik.values.recordId ?

**Pull request merged**